modules/renter/contractor/contracts.go

```diff
@@ -185,9 +185,14 @@ func (c *Contractor) managedNewContract(host modules.HostDBEntry, contractFundin
 	// Add a mapping from the contract's id to the public key of the host.
 	c.mu.Lock()
 	c.contractIDToPubKey[contract.ID] = contract.HostPublicKey
+	_, exists := c.pubKeysToContractID[string(contract.HostPublicKey)]
 	c.pubKeysToContractID[string(contract.HostPublicKey.Key)] = contract.ID
 	c.mu.Unlock()
 
+	if exists {
+		build.Critical("We are forming a new contract but the host's pubKey is already mapped to a filecontract's id")
```

Comment:
I don't know if this needs to be a `build.Critical`, I feel like just returning and error and logging it is fine.